source/pick-from-possibly-undefined.d.ts

```diff
@@ -0,0 +1,23 @@
+/**
+It allows to `Pick` properties from a type that may be undefined
+
+@example:
+```
+type User = {
+    id: number;
+    name: string;
+    email: string;
+} | undefined;
+
+type UserWithId = PickFromPossiblyUndefined<User, 'id' | 'name'>;
+```
+
+// Results in: UserWithId = { id: number, name: string }
+
+@category Object
+ **/
+export type PickFromPossiblyUndefined<Type, Props extends keyof NonNullable<Type>> = Type extends undefined
```

Comment:
What is the different with 
```typescript
type PickFromPossiblyUndefined<Type, Props extends keyof NonNullable<Type>> = Pick<NonNullable<Type>, Props>
```

Reply:
it won't work the same (tests would crash).